examples/elasticsearch-operator/elasticsearch-agent.yaml

```diff
@@ -0,0 +1,40 @@
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: elasticsearch-agent-sample
+spec:
+  components:
+    - type: elasticsearch-agent
+      name: elasticsearch-agent
```

Comment:
the agent should be a trait, to collect logs from component.

BTW, which mode does the agent run?

If it's daemonset, only one agent should be created, the rest should be a new config merged to the agent configuration.

If it's sidecar, it should patch to the pods of workload.

We can't create lots of agents if it's running as daemonset.

Reply:
It's daemonset.

Reply:
Done :+1